cmd/kubeseal/main.go

```diff
@@ -236,17 +252,75 @@ func validateSealedSecret(in io.Reader, namespace, name string) error {
 		Resource("services").
 		SubResource("proxy").
 		Name(net.JoinSchemeNamePort("http", name, "")).
-		Suffix("/v1/verify")
+		Suffix("/v1/rotate")
 
 	req.Body(content)
 	res := req.Do()
 	if err := res.Error(); err != nil {
 		if status, ok := err.(*k8serrors.StatusError); ok && status.Status().Code == http.StatusConflict {
-			return fmt.Errorf("Unable to decrypt sealed secret")
+			return fmt.Errorf("Unable to rotate secret")
 		}
-		return fmt.Errorf("Error occurred while validating sealed secret")
+		return fmt.Errorf("Error occurred while rotating secret")
+	}
+	body, err := res.Raw()
+	if err != nil {
+		return err
 	}
+	ssecret := &ssv1alpha1.SealedSecret{}
+	if err = json.Unmarshal(body, ssecret); err != nil {
+		return err
+	}
+	ssecret.SetCreationTimestamp(metav1.Time{})
+	ssecret.SetDeletionTimestamp(nil)
+	ssecret.Generation = 0
+	if err = sealedSecretOutput(out, codecs, ssecret); err != nil {
+		return err
+	}
+	return nil
+}
+
+func sealedSecretOutput(out io.Writer, codecs runtimeserializer.CodecFactory, ssecret *ssv1alpha1.SealedSecret) error {
+	var contentType string
+	switch strings.ToLower(*outputFormat) {
+	case "json", "":
+		contentType = runtime.ContentTypeJSON
+	case "yaml":
+		contentType = "application/yaml"
+	default:
+		return fmt.Errorf("unsupported output format: %s", *outputFormat)
+	}
+	prettyEnc, err := prettyEncoder(codecs, contentType, ssv1alpha1.SchemeGroupVersion)
+	if err != nil {
+		return err
+	}
+	buf, err := runtime.Encode(prettyEnc, ssecret)
+	if err != nil {
+		return err
+	}
+	out.Write(buf)
+	fmt.Fprint(out, "\n")
+	return nil
+}
 
+func generateKey(namespace, name string) error {
+	conf, err := clientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+	restClient, err := corev1.NewForConfig(conf)
+	if err != nil {
+		return err
+	}
+	req := restClient.RESTClient().Post().
+		Namespace(namespace).
+		Resource("services").
+		SubResource("proxy").
+		Name(net.JoinSchemeNamePort("http", name, "")).
+		Suffix("/v1/keygen")
```

Comment:
Did we remove this endpoint?

Reply:
We did, I forgot to remove this function from kubeseal, thanks for pointing it out.

Reply:
We did, I forgot to remove this function from kubeseal, thanks for pointing it out.